pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -229,7 +229,7 @@ def _evalute_stopping_criteria(self, current: torch.Tensor) -> Tuple[bool, str]:
                 f" {self.monitor} = {current} {self.order_dict[self.mode]} {self.divergence_threshold}."
                 " Signaling Trainer to stop."
             )
-        elif self.monitor_op(current - self.min_delta, self.best_score):
+        elif self.monitor_op(current - self.min_delta, self.best_score.to(trainer.lightning_module.device)):
```

Comment:
TBH: I don't get why this is even necessary. We usually don't remove them from the device (except explicitly specifying this with `move_metrics_to_cpu=True`. So I think we should rather investigate where this happens.

Reply:
Hey @kaushikb11, the tensors are being moved to cpu there: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/logger_connector/result.py#L398.

I think this line was added to enable logger to log values as they need those tensors to be on `cpu`. I guess we could move this function to `log_metrics` in LoggerConnector with apply_to_collection. Mind giving it a try ?

Best,
T.C

Reply:
Hey @kaushikb11. Mind giving a try to removing those lines in `ResultCollection.log` ?

Reply:
This seems to happen for

which hasn't been moved to the TPU.
 
```
self.best_score = torch_inf if self.monitor_op == torch.lt else -torch_inf
```

Thanks @kaushikb11 for the clarification.